mds/apis/utils.py

```diff
@@ -183,3 +180,16 @@ def get_default_response_serializer(self):
             return body_override
 
         return self.get_view_serializer("response")
+
+
+# date & time #########################################################
+
+
+def datetime_to_millis(value: datetime.datetime):
+    td = value - datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc)
+    return int(td.total_seconds() * 1000 + td.microseconds)
```

Comment:
Simplement `int(value.timestamp() * 1000)`, ça ne marche pas ? Ça devrait, même avec une datetime timezone-aware

Reply:
until 2038 this should indeed work.

Reply:
there is a unicity constraint on timestamp to avoid duplicating provider and agency information - hence we try to keep the timestamp as precise as possible (to the millisecond)